analyzers/src/SonarAnalyzer.Common/Helpers/CodeCopiedFromAttribute.cs

```diff
@@ -0,0 +1,42 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Helpers;
+
+public enum TrustedCodeSource
+{
+    Roslyn,
+    StyleCop
+}
+
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface | AttributeTargets.Method | AttributeTargets.Constructor | AttributeTargets.Property,
```

Comment:
Given that `AttributeTargets` is a `Flags` enum, what about not restricting it, and use `AttributeTargets.All`?
Less precise maybe, but doesn't require to list all possible targets of copy.

Reply:
It needs to be understood by the coverage tool but at the same time, we already copied enums to our code base and it might be useful to have that as a valid target. I don't think we should allow Assembly/Module though. Others, like fields/events, seem to be granular to me.